File: lib/features/auth/domain/user_role.dart
enum UserRole {
  admin,
  approval,
  editor,
  crudUser;

  String get label {
    switch (this) {
      case UserRole.admin:
        return 'Super Admin';
      case UserRole.approval:
        return 'Approval';
      case UserRole.editor:
        return 'Editor';
      case UserRole.crudUser:
        return 'CRUD User';
    }
  }

  String l10nKey() {
    switch (this) {
      case UserRole.admin:
        return 'role_admin';
      case UserRole.approval:
        return 'role_approval';
      case UserRole.editor:
        return 'role_editor';
      case UserRole.crudUser:
        return 'role_crud_user';
    }
  }

  static UserRole fromString(String value) {
    final normalized =
        value.trim().toLowerCase().replaceAll(RegExp(r'[\s_-]'), '');
    switch (normalized) {
      case 'admin':
      case 'globaladmin':
        return UserRole.admin;
      case 'approval':
      case 'genderadmin':
        return UserRole.approval;
      case 'editor':
        return UserRole.editor;
      case 'cruduser':
        return UserRole.editor;
      default:
        return UserRole.editor;
    }
  }

  String toFirestore() {
    switch (this) {
      case UserRole.admin:
        return 'admin';
      case UserRole.approval:
        return 'approval';
      case UserRole.editor:
      case UserRole.crudUser:
        return 'editor';
    }
  }
}

File: app/src/main/java/com/example/salik_management_system/auth/domain/UserRole.kt
package com.example.salik_management_system.auth.domain

enum class UserRole {
    Admin,
    Approval,
    Editor;

    val label: String
        get() = when (this) {
            Admin -> "Super Admin"
            Approval -> "Approval"
            Editor -> "Editor"
        }

    fun toFirestore(): String = when (this) {
        Admin -> "admin"
        Approval -> "approval"
        Editor -> "editor"
    }

    companion object {
        fun fromString(value: String): UserRole {
            val normalized = value.trim().lowercase().replace(Regex("[\\s_-]"), "")
            return when (normalized) {
                "admin", "globaladmin" -> Admin
                "approval", "genderadmin" -> Approval
                "editor", "cruduser" -> Editor
                else -> Editor
            }
        }
    }
}
